src/signer/canister/src/lib.rs

```diff
@@ -128,21 +133,42 @@ async fn sign_prehash(prehash: String) -> String {
 
 /// Returns the Bitcoin address of the caller.
 #[update(guard = "caller_is_not_anonymous")]
-async fn caller_btc_address(network: BitcoinNetwork) -> String {
-    bitcoin_utils::public_key_to_p2pkh_address(
-        network,
-        &bitcoin_utils::ecdsa_pubkey_of(&ic_cdk::caller()).await,
-    )
+async fn caller_btc_address(
+    params: GetAddressRequest,
+) -> Result<GetAddressResponse, GetAddressError> {
```

Comment:
I would like to propose having all Bitcoin API method names starting with `btc_` and similarly with other chains.  This way, when working with Bitcoin (say) all the methods applicable to bitcoin are grouped together in the candid file and in the rustdoc.  What do you think?  Would this be reasonably easy to do?

Payment will have to be added to this method, but I can presumably add payment errors as `GetAddressError` variants, so that is fine!  I can add that in a later PR.

Reply:
Discussed - the btc prefix can be added in another PR.

Reply:
btc prefix added in this PR which already has breaking change in the interface.